middlewares.js

```diff
@@ -0,0 +1,26 @@
+const R = require('ramda');
+const {jsonOr} = require('./conversions');
+const { vsprintf } = require('sprintf-js')
+
+const withDataOr = R.curry((key, defaultValue, req, res, next) => {
+    req[key] = jsonOr(req.get(key), defaultValue);
+    next();
+});
+
+const withLang = R.curry((allowed, defaultLang, key, req, res, next) => {
+    const raw = (req.query && req.query.locale || req.headers && req.headers[key] || '').toLowerCase().split(/[-_]/g);
+    req.lang = raw && allowed.includes(raw[0]) && raw[0] || defaultLang;
+    next()
+});
+
+//Requires withLang
+const withTranslate = R.curry((translations, req, res, next) => {
```

Comment:
I don't think we should allow to have the `withTranslate` middleware that can be applied without the `withLang`

Reply:
I combined `withTranslate` with `withLang` and cleared out most of the `unsignedIntOr` functions and left others like `jsonOr` since it feels like an `express util`. Plus some tests.